elastic/kibana
# [Discover] Extend Elasticsearch query rule with search source based data fetching

## Summary

Closes #117532

This PR extends Elasticsearch query rule to enable the creation of an alert rule in Discover via `Search source`. See all details in the linked issue.

### Test
- Open Discover
- Click `Alerts` in top nav menu
- Fill the inputs and create the alert
- Open received alert and follow the link provided
- It should navigate to the documents that triggered the alert

**Alert creation sidebar**

![D2C3D8B8-B053-4272-8B62-999506E58E81_1_105_c](https://user-images.githubusercontent.com/39378793/158744772-006d6832-1cf8-4ea9-a4da-7c466e4685cd.jpeg)


**Tool tip with explanation about missed time field in data view**

![D02E160D-BEEC-4D0F-83B2-14F5BCEE84CC](https://user-images.githubusercontent.com/39378793/155537696-04d78f9f-d3a6-409a-8fb5-af0b1100620a.jpeg)

**Toast info notification about possible difference to the documents that triggered alert when user follows link provided by rule connector**

![E5298C4E-7756-4266-9604-CFABFE482B2C_4_5005_c](https://user-images.githubusercontent.com/39378793/156154084-72a15321-bf90-4358-9e7d-87b04daa74d6.jpeg)



### Checklist

Delete any items that are not applicable to this PR.
